src/averageAges.js

```diff
@@ -37,7 +40,20 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const filterWom = people.filter(wom => withChildren
+    ? findMomByName(people, wom) : wom.sex === 'f');
+
+  const ageWom = filterWom.map((age) => differenceAge(age));
+
+  return calculatePeoplesAge(ageWom);
+}
+
+function findMomByName(people, wom) {
+  return people.some(nameMother => wom.name === nameMother.mother);
```

Comment:
The function name `findMomByName` might be confusing as it suggests that the function returns a mom object, but in reality, it returns a boolean. Consider renaming it to something like `isMother`.